src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs

```diff
@@ -9,218 +9,177 @@ namespace Microsoft.CodeAnalysis.Sarif.Visitors
                     }
                 }
 
-                if (node.Tags.Count > 0)
+                if (v1FileData.Tags.Count > 0)
                 {
-                    fileData.Tags.UnionWith(node.Tags);
+                    fileData.Tags.UnionWith(v1FileData.Tags);
                 }
             }
 
             return fileData;
         }
 
-        public static LogicalLocation CreateLogicalLocation(LogicalLocationVersionOne node)
+        public static LogicalLocation CreateLogicalLocation(LogicalLocationVersionOne v1LogicalLocation)
         {
             LogicalLocation logicalLocation = null;
 
-            if (node != null)
+            if (v1LogicalLocation != null)
             {
                 logicalLocation = new LogicalLocation
                 {
-                    Kind = node.Kind,
-                    Name = node.Name,
-                    ParentKey = node.ParentKey
+                    Kind = v1LogicalLocation.Kind,
+                    Name = v1LogicalLocation.Name,
+                    ParentKey = v1LogicalLocation.ParentKey
                 };
             }
 
             return logicalLocation;
         }
 
-        public static Hash CreateHash(HashVersionOne node)
+        public static Hash CreateHash(HashVersionOne v1Hash)
         {
             Hash hash = null;
 
-            if (node != null)
+            if (v1Hash != null)
             {
                 string algorithm;
-                if (!s_AlgorithmKindNameMap.TryGetValue(node.Algorithm, out algorithm))
+                if (!SarifTransformerUtilities.AlgorithmKindNameMap.TryGetValue(v1Hash.Algorithm, out algorithm))
                 {
-                    algorithm = node.Algorithm.ToString().ToLowerInvariant();
+                    algorithm = v1Hash.Algorithm.ToString().ToLowerInvariant();
                 }
 
                 hash = new Hash
                 {
                     Algorithm = algorithm,
-                    Value = node.Value
+                    Value = v1Hash.Value
                 };
             }
 
             return hash;
         }
 
-        public static Rule CreateRule(RuleVersionOne node)
+        public static Rule CreateRule(RuleVersionOne v1Rule)
         {
             Rule rule = null;
 
-            if (node != null)
+            if (v1Rule != null)
             {
                 rule = new Rule
                 {
-                    Id = node.Id,
-                    MessageStrings = node.MessageFormats,
-                    Properties = node.Properties
+                    Id = v1Rule.Id,
+                    MessageStrings = v1Rule.MessageFormats,
+                    Properties = v1Rule.Properties
                 };
 
-                if (node.Configuration != RuleConfigurationVersionOne.Unknown &&
-                    node.DefaultLevel != ResultLevelVersionOne.Default)
+                if (v1Rule.Configuration == RuleConfigurationVersionOne.Enabled &&
```

Comment:
This needs to be `||`, not `&&`. You want to emit the object if _either_ of the properties is non-default.

Reply:
Grr, sorry

Reply:
So I'm getting an empty value when defaultLevel is "note" (converted from "pass")

Reply:
You're saying that when you have a v1 file with `rule.defaultLevel == "note"`, and you convert it to v2, that the v2 file does _not_ contain a `rule.configuration` object? That behavior doesn't match the code (or at least, it doesn't match the code as it should be written):
```
if (v1Rule.Configuration == RuleConfigurationVersionOne.Enabled ||
    v1Rule.DefaultLevel != ResultLevelVersionOne.Warning)
{
    // Create the Configuration object.
}
```
How sure are you of this? Like, you saved the file, and rebuilt, and ran the test, and you got that result?

Reply:
When the v1 value is "pass", it gets converted to v2 "note" and nothing is emitted. I'm watching the value get set in the debugger.

Reply:
When the debugger hits the conditional, does the conditional return `true`? Do you get into the code block that creates the `RuleConfiguration` object?

I'm gonna ask you to investigate this one.

Reply:
OK I wrote a whole big thing but I'll give you the abridged version: whatever enum member has value 0 will not be emitted. Even the default ("warning") emits as long as it's not the first member.  I'm happy to explain all of the experiments I performed.

This isn't caught by the current test because the values in the OM are correct (big advantage of comparing json!),

Reply:
Ok. It seems to me that we want to always emit the `ruleConfiguration.defaultLevel` property, even if it is the "0" value.

Is there a codegen hint that says "don't emit the `EmitDefaultValue` attribute"?

Reply:
Sorry, I mean: "if we are going to emit the `ruleConfiguration` object _at all_, we should always emit its `defaultLevel` property". We should still test for the "all defaults" case so that we can decide not to emit the  `ruleConfiguration` object at all.

Reply:
Right. I think all we need is to add a "zeroValueName": "Default" hint. Result.Level has this.

Reply:
Ah, good.

Reply:
I used "None" instead of "Default", that ok?

Reply:
Sure, ok, `None`.

Reply:
Yeah that was fixed after my last push. Current code:
RuleConfigurationDefaultLevel level = SarifTransformerUtilities.CreateRuleConfigurationDefaultLevel(v1Rule.DefaultLevel);

if (v1Rule.Configuration == RuleConfigurationVersionOne.Enabled ||
    level != RuleConfigurationDefaultLevel.Warning)
{
    rule.Configuration = new RuleConfiguration
    {
        DefaultLevel = level,
        Enabled = v1Rule.Configuration == RuleConfigurationVersionOne.Enabled
    };
}

Reply:
Ok.